hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts

```diff
@@ -1,32 +1,95 @@
-import { createEntityAdapter, createSelector, createSlice } from "@reduxjs/toolkit";
-import { Order } from "api/types";
-import { RootState } from "slices/store";
+import {
+    createAsyncThunk,
+    createEntityAdapter,
+    createSelector,
+    createSlice,
+} from "@reduxjs/toolkit";
+import { APIListResponse, Order, OrderInTable, ReturnOrderInTable } from "api/types";
+import { AppDispatch, RootState } from "slices/store";
+import { get } from "api/api";
+import { teamOrderListSerialization } from "api/helpers";
 
 interface OrderExtraState {
     isLoading: boolean;
     error: string | null;
+    next: string | null;
+    hardwareInOrders: number[] | null;
+    returnedOrders: ReturnOrderInTable[];
+    checkedOutOrders: OrderInTable[];
 }
 
 const extraState: OrderExtraState = {
     isLoading: false,
     error: null,
+    next: null,
+    hardwareInOrders: null,
+    returnedOrders: [],
+    checkedOutOrders: [],
 };
 
-const orderAdapter = createEntityAdapter<Order>();
+const pendingOrderAdapter = createEntityAdapter<OrderInTable>();
 
 export const orderReducerName = "participantOrder";
-export const initialState = orderAdapter.getInitialState(extraState);
+export const initialState = pendingOrderAdapter.getInitialState(extraState);
 export type OrderState = typeof initialState;
 
 interface RejectValue {
     status: number;
     message: any;
 }
 
+export const getTeamOrders = createAsyncThunk<
+    APIListResponse<Order>,
+    undefined,
+    { state: RootState; rejectValue: RejectValue; dispatch: AppDispatch }
+>(`${orderReducerName}/geTeamOrders`, async (_, { rejectWithValue }) => {
```

Comment:
this should be getTeamOrders on line 45 right?

Reply:
yes, thanks for catching!